### ISSUE TITLE
SSL connection to AWS RDS

### ISSUE THREAD

**Issue by garethmdavies** [r0man/sqlingvo.node#1]
Hi, I'm trying to connect to a postgres RDS database over SSL using sqlingvo.node but hitting an error executing my query. My use case is a clourescript aws lambda that connect to a RDS postgres db. The problem I get is that when I execute my query i get "TypeError: Pp.a(...).Xd is not a function" returned.

Using a local database everything works fine however for RDS I need to add ssl=true and sslmode=verify-ca or sslmode=verify-full.

Any idea how to configure sqlinvgo.node to work over ssl with a cert.

**Comment by r0man** [comment]
@taffowl No, I haven't used sqlingvo.node over SSL. Are you sure the problem is with SSL, and not the advanced optimization settings? This `Pp.a(...).Xd` looks like a problem with renaming to me.

**Comment by r0man** [comment]
@taffowl Did you try adding ssl=true etc, to the query params of the connection url?

**Comment by garethmdavies** [comment]
Yes I'm using ssl=true but you may be onto something with the optimizations. I'm using this aws-library (https://github.com/nervous-systems/cljs-lambda) to build my lambda which defaults to advanced optimizations. I've tried playing around with different cljsbuild settings but can't get a working lambda in aws without the advanced mode. I'll try packaging a lambda from scratch to rule out the optimization mode.

**Comment by garethmdavies** [comment]
Just an update I did get this working using simple optimisation. The ssl was a red herring which I confirmed by writing a pure node aws lambda to talk to RDS.

I ended up forking cljs-lambda and adding simple optimisation support and it works great now with sqlingvo.node. The one thing that caught me out was the libpq binaries that were being included by my mac build. Removing the libpq files from my lambda package removed some invalid e.g. Header errors I was also getting.
